### ISSUE TITLE
[BUG] koordlet "invalid memory address or nil pointer dereference" error

### ISSUE THREAD

**Issue by nashtsai** [koordinator-sh/koordinator#1797]
**What happened**:
koordlet "invalid memory address or nil pointer dereference" error

```
I1227 18:00:35.775470 3553094 rule.go:99] rule CPUSetAllocator is updated, run update callback for all 33 pods
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a74d4e]

goroutine 1181 [running]:
github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/hooks/cpuset.(*cpusetPlugin).SetContainerCPUSet(0xc000757b18?, {0x28e3000?, 0xc000757b18})
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/hooks/cpuset/cpuset.go:123 +0x7ce
github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/hooks/cpuset.(*cpusetPlugin).ruleUpdateCb(0xc000f67770, {0xc0017ce6c0, 0x21, 0x2?})
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/hooks/cpuset/rule.go:155 +0x595
github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/rule.(*Rule).runUpdateCallbacks(0xc0005fa460, {0xc0017ce6c0, 0x21, 0x21})
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/rule/rule.go:61 +0xbe
github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/rule.UpdateRules(0x2, {0x23b9440, 0xc0006fa000}, {0xc0017ce6c0, 0x21, 0x21})
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/rule/rule.go:100 +0x41c
github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer/impl.(*callbackRunner).runCallbacks(0xc0006142e0, 0x2, {0x23b9440, 0xc0006fa000})
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer/impl/callback_runner.go:99 +0x3c3
github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer/impl.(*callbackRunner).Start.func1()
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer/impl/callback_runner.go:114 +0x105
created by github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer/impl.(*callbackRunner).Start
	/go/src/github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer/impl/callback_runner.go:106 +0x70
(venv-3.11.4)
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
Testing nodemetrics only feature, and only koordlet pod running on Master nodes has above error.

**Environment**:
- App version: 1.3.0
- Kubernetes version (use `kubectl version`): v1.19
- Install details (e.g. helm install args): Only install koordlet & CRDs: [nodemetrics.slo.koordinator.sh, nodeslos.slo.koordinator.sh, noderesourcetopologies.topology.node.k8s.io]
- Node environment (for koordlet/runtime-proxy issue):
  - Containerd/Docker version: `containerd://1.6.21`
  - OS version: Debian9
  - Kernal version: `5.4.210` & `4.19.117`
  - Cgroup driver: cgroupfs/systemd
- Others:

**Comment by saintube** [comment]
@nashtsai Thanks for your feedback, and we're sorry for the mistake. It is a bug that has been fixed by #1638. The bugfix will be released in the coming v1.4.0.

**Comment by saintube** [comment]
/close
Duplicate #1745.
Please feel free to reopen the issue if there are still concerns.
